frappe/search/website_search.py

```diff
@@ -85,16 +87,23 @@ def parse_result(self, result):
 		)
 
 
-def slugs_with_web_view():
+def slugs_with_web_view(_items_to_index):
 	all_routes = []
 	filters = { "has_web_view": 1, "allow_guest_to_view": 1, "index_web_pages_for_search": 1}
-	fields = ["name", "is_published_field"]
+	fields = ["name", "is_published_field", 'website_search_field']
 	doctype_with_web_views = frappe.get_all("DocType", filters=filters, fields=fields)
 
 	for doctype in doctype_with_web_views:
 		if doctype.is_published_field:
-			routes = frappe.get_all(doctype.name, filters={doctype.is_published_field: 1}, fields="route")
-			all_routes += [route.route for route in routes]
+			docs = frappe.get_all(doctype.name, filters={doctype.is_published_field: 1}, fields=["route", doctype.website_search_field, 'title'])
```

Comment:
the value of `website_search_field` needs to be validated to avoid failure if the user enters random text.

Reply:
added validations